windows-agent/internal/daemon/netwatcher_test.go

```diff
@@ -0,0 +1,141 @@
+package daemon_test
+
+import (
+	"context"
+	"errors"
+	"maps"
+	"testing"
+	"time"
+
+	"github.com/canonical/ubuntu-pro-for-wsl/windows-agent/internal/daemon"
+	"github.com/canonical/ubuntu-pro-for-wsl/windows-agent/internal/daemon/daemontestutils"
+	"github.com/canonical/ubuntu-pro-for-wsl/windows-agent/internal/daemon/netmonitoring"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSubscribe(t *testing.T) {
+	t.Parallel()
+
+	mockError := errors.New("mock error")
+
+	before := map[string]string{
+		uuid.New().String(): "conn0",
+		uuid.New().String(): "conn1",
+		uuid.New().String(): "conn2",
+	}
+
+	after := map[string]string{
+		uuid.New().String(): "new",
+		"not a guid":        "yet_another_new",
+	}
+
+	maps.Copy(after, before)
+
+	testcases := map[string]struct {
+		initError             error
+		listDevicesError      error
+		listDevicesAfterError error
+		getConnNameError      error
+		waitError             error
+
+		ctxCancel           bool
+		startWithNoAdapters bool
+		devicesUnchanged    bool
+
+		wantErr        bool
+		wantNoCallback bool
+		wantName       string
```

Comment:
nitpick: I would put `wantName` first in the want and `wantErr` last to get the "normal happy flow" first :)